### ISSUE TITLE
Not working with Ubuntu 14.10

### ISSUE THREAD

**Issue by simon-bourne** [tailhook/vagga#17]
Hi, I couldn't get Vagga working with Utopic:

containers:
    build:
        setup:
        - !Ubuntu utopic
commands:
    run: !Command
        container: build
        run: "$@"

It's trying to download http://cdimage.ubuntu.com/ubuntu-core/utopic/daily/current/utopic-core-amd64.tar.gz, which doesn't exist. If the URLs were of the form http://cdimage.ubuntu.com/ubuntu-core/releases/14.10/release/ubuntu-core-14.10-core-amd64.tar.gz, they'd work with all releases from 12.04. Unfortunately this only works with version numbers (due to the version number in the filename), which would break existing config. Although a map of existing releases to version numbers would ensure backwards compatibility. Version numbers also have the advantage that point releases can be specified like 14.04.2. The directory seems to be named 14.04.2 and the file named just ubuntu-core-14.04-core-amd64.tar.gz, without the ".2".

**Comment by tailhook** [comment]
Well. Right. The difference is however, that we currently use daily builds (i.e. they are updated more regularly, except our local caching). But there are daily builds only for LTS versions. We have few options:

1. Map names to (stable) versions
2. Use release for version numbers and daily builds for named (LTS) versions
3. Introduce another command i.e. `!UbuntuStable 14.10` for other versions (better name?)

Would you like to contribute a patch?

**Comment by simon-bourne** [comment]
Ah, I see. I'd favour option 3 as it makes the distinction between daily and release versions. I was having trouble getting it to build yesterday, but I'll take another look at adding a patch.

**Comment by tailhook** [comment]
> I was having trouble getting it to build yesterday, but I'll take another look at adding a patch.

Any specific error? The best way to build vagga is using vagga itself :)

**Comment by simon-bourne** [comment]
I had forked an old fork from another repo accidentally... Once I'd forked the correct repo it all worked great! I'm adding code to allow the following:

containers:
    build:
        setup:
        - !UbuntuRelease { name: utopic, version: 14.10 }

It's not ideal that you have to specify name and version, but version is required for download and name is required for enabling universe. I couldn't find a good way of getting name from version. There's /etc/os-release, but it didn't look like there was a robust way to get name from version. It has lines like:

VERSION="14.04.2 LTS, Trusty Tahr"
 or
VERSION="14.10 (Utopic Unicorn)"

vagga was very useful for figuring that out!

Also, having UbuntuRelease as a struct seems more extensible. I've chosen the name "release" as that seems to be the common terminology. What do you reckon?

**Comment by tailhook** [comment]
Looks ok. Just one question. Do you think parsing `/etc/apt/sources.list` for getting version name would be wrong (if used solely for enabling universe)?

I think format is easy to parse, just not sure if it's a reasonable idea.

**Comment by simon-bourne** [comment]
Makes me a little nervous, but can't immediately think of anything wrong with it, as long as it was parsed straight after download. I'll take a look at what the initial /etc/apt/ tree looks like on different versions.

**Comment by cablehead** [comment]
Another possibility is uname:

```

andy@hop:~$ uname -a

Linux hop 3.16.0-30-generic #40~14.04.1-Ubuntu SMP Thu Jan 15 17:43:14 UTC
2015 x86_64 x86_64 x86_64 GNU/Linux

```

**Comment by simon-bourne** [comment]
Thanks, but it's the codename (utopic, trusty etc) that I'm after. I couldn't see anything that would coax that out of uname, but let me know if I've missed something. I think I have a way forward now in /etc/lsb-release, which seems to be present on the original cd image.

**Comment by tailhook** [comment]
Seems to be fixed in #18. I'll cut the release a little bit later, as I want to have SubConfig functionality stabilized.
